### ISSUE TITLE
Level select remaining problems

### ISSUE THREAD

**Issue by clickyquack** [darienyoder/software-engineering-team-beta#86]
1. Tests need to be written for it
2. The camera change settings currently do not work in level select mode
3. Somehow a regression occurred and it no longer returns to the main menu once a level is finished in level select mode
4. Once you finish a level from level select mode, you can't enter a new level from level select mode (interestingly, you can play through normal mode)

**Comment by clickyquack** [comment]
For #1, I opened a separate issue: https://github.com/darienyoder/software-engineering-team-beta/issues/87
For #2-3, I fixed them here: https://github.com/darienyoder/software-engineering-team-beta/tree/level-select-2
For #4, I discovered a new strange bug and after trying for over an hour I wasn't able to pinpoint what was causing it. But, in theory, if that bug is fixed, it will solve the problem. Currently, checking whether you clicked a level square on the level select screen is determined by manually checking the cursor position against the bounds of the level square. Normally, the mouse.x and mouse.y variables are determined by how far horizontally and vertically the cursor is from the top left corner of the browser window, respectively. I found out, however, that upon starting a level, it instead measures it relative to how far the cursor is from the top left corner of the starting area of the level. This wouldn't be a problem, if it weren't for the fact that it continues to measure the cursor position relative to where the top left corner of the level _was_ once you go back to the level select screen from the main menu after finishing the level. If, however, you line up the new mouse coordinates with where a level square is supposed to be, it opens the respective level. (e.g. suppose the level square for level 1 can be originally clicked at x=50, y=50. If you move the cursor to "x=50, y=50" after the cursor measurement shifts, it opens level 1 still.). I have no idea why this is happening, or how to fix it. I've written a branch to show cursor coordinates during level selection and gameplay here: https://github.com/darienyoder/software-engineering-team-beta/tree/cursor-coordinates-bug.

**Comment by clickyquack** [comment]
I spent multiple hours trying to craft an elaborate solution for this and then I discovered that for some reason using the variables "mouseX" and "mouseY" instead of "mouse.x" and "mouse.y" make everything work as intended. I have no idea why they're different in the first place. Anyway, it's fixed here: https://github.com/darienyoder/software-engineering-team-beta/tree/level-select-2. Also, I deleted the "cursor coordinates bug" branch as it's no longer needed.
